UPGRADE-3.x.md

```diff
@@ -1,6 +1,28 @@
 UPGRADE 3.x
 ===========
 
+## Added constant for show/list field types and deprecated some field types.
```

Comment:
We usually only add the deprecated part, but it certainly does not hurt to do what you did :slightly_smiling_face:

Reply:
Yeah I know. I wanted to promote the new constants, since I think it's always better to use constant rather than string.

I made others changes :)